docroot/modules/custom/prisoner_hub_entity_access/tests/src/ExistingSite/PrisonerHubQueryAccessTest.php

```diff
@@ -0,0 +1,228 @@
+    // Create some nodes that have no values.
+    for ($i = 0; $i < 5; $i++) {
+      $values = [
+        'type' => 'page',
+      ];
+      $this->createNode($values)->uuid();
+    }
+    $this->assertJsonResponseNodes([]);
+  }
+
+  /**
+   * Test that content appears when tagged with a prison (but no category).
+   */
+  public function testContentTaggedWithPrisonButNoCategory() {
+    $nodes_to_check = [];
+    for ($i = 0; $i < 5; $i++) {
+        $values = [
+          'type' => 'page',
+          $this->prisonFieldName => [
+            ['target_id' => $this->prisonTerm->id()]
+          ],
+        ];
+      $nodes_to_check[] = $this->createNode($values)->uuid();
+    }
+
+    // Also create some content tagged with a different prison.
+    for ($i = 0; $i < 5; $i++) {
+      $values = [
+        'type' => 'page',
+        $this->prisonFieldName => [
+          ['target_id' => $this->anotherPrisonTerm->id()],
+        ],
+      ];
+      $this->createNode($values);
+    }
+
+    $this->assertJsonResponseNodes($nodes_to_check);
+  }
+
+  /**
+   * Test that content appears when tagged with a category (but no prison).
+   */
+  public function testContentTaggedWithCategoryButNoPrison() {
+    $nodes_to_check = [];
+    for ($i = 0; $i < 5; $i++) {
+      $values = [
+        'type' => 'page',
+        $this->prisonCategoryFieldName => [
+          ['target_id' => $this->prisonCategoryTerm->id()],
+        ],
+      ];
+      $nodes_to_check[] = $this->createNode($values)->uuid();
+    }
+
+    // Also create some content tagged with a different prison category.
+    for ($i = 0; $i < 5; $i++) {
+      $values = [
+        'type' => 'page',
+        $this->prisonCategoryFieldName => [
+          ['target_id' => $this->anotherPrisonCategoryTerm->id()],
+        ],
+      ];
+      $this->createNode($values)->uuid();
+    }
+
+    $this->assertJsonResponseNodes($nodes_to_check);
+  }
+
+  /**
+   * Test that content appears when tagged with a category and a prison.
+   */
+  public function testContentTaggedWithPrisonAndCategory() {
+    $nodes_to_check = [];
+    for ($i = 0; $i < 5; $i++) {
+      $values = [
+        'type' => 'page',
+        $this->prisonFieldName => [
+          ['target_id' => $this->prisonTerm->id()],
+        ],
+        $this->prisonCategoryFieldName => [
+          ['target_id' => $this->prisonTerm->id()],
+        ],
+      ];
+      $nodes_to_check[] = $this->createNode($values)->uuid();
+    }
+
+    // Also create some content tagged with a different category and prison.
+    for ($i = 0; $i < 5; $i++) {
+      $values = [
+        'type' => 'page',
+        $this->prisonFieldName => [
+          ['target_id' => $this->anotherPrisonTerm->id()]
+        ],
+        $this->prisonCategoryFieldName => [
+          ['target_id' => $this->anotherPrisonCategoryTerm->id()],
+        ],
+      ];
+      $this->createNode($values)->uuid();
+    }
+
+    $this->assertJsonResponseNodes($nodes_to_check);
+  }
```

Comment:
Should we add a test for when content is tagged the same category but a different prison (asserting that it's not shown)?

Reply:
So..... I was wondering what the requirement was around this, whether the prison field should _override_ the prison category field (i.e. if a prison is set, then ignore anything from the prison category field).  
Or whether they should be independent (i.e. show content if either it matches the prison or prison category field).  
I opted for the latter, which I think is a simpler solution, and means that we are more likely to be showing content rather than hiding it.
I also feel it is more easier to understand to the CMS user.  Otherwise they might select a prison category, expecting to see their content appear, and not notice that there was a prison selected, and suspect something was broken when they couldn't see their content.

But happy to discuss this further.